src/shared/tpm2-util.c

```diff
@@ -1519,7 +1516,13 @@ static int tpm2_make_encryption_session(
 
                 CLEANUP_ERASE(auth);
 
-                hash_pin(pin, strlen(pin), &auth);
+                r = tpm2_digest_init_buffer(
+                                TPM2_ALG_SHA256,
+                                tpm2_digest_from_auth(&auth),
+                                (uint8_t*) pin,
```

Comment:
this function should just take (void*) for this, so that the cast here is not necessary.

(the cast is particularly ugly, since it drops the "const" for no reason)

Reply:
dropped the cast here, though the tpm2_digest_buffer() define does still need to cast to (void*) because it's generating an intermediate struct iovec[] (with one element), and struct iovec.iov_base isn't const so it needs to cast to (void*)